WordPress/src/main/java/org/wordpress/android/ui/sitecreation/SiteCreationMainVM.kt

```diff
@@ -139,7 +143,7 @@ class SiteCreationMainVM @Inject constructor(
         val stepCount = wizardManager.stepsCount
         val firstStep = stepPosition == 1
         val lastStep = stepPosition == stepCount
-        val singleInBetweenStepDomains = wizardManager.stepsCount == 3 && step.name == DOMAINS.name
+        val singleInBetweenStepDomains = step.name == DOMAINS.name
```

Comment:
> I don't understand what was the reason to also check for `wizardManager.stepsCount == 3`

Not quite sure either, but it seems to have been a part of this issue: https://github.com/wordpress-mobile/WordPress-Android/issues/11420 which interestingly enough, was a long ago removal of similar screens that we are now adding (though the designs and features we are planning for this are improved since then). My guess is that the extra condition was a safety feature during the removal of steps :man_shrugging: .

Tbh, I find the current logic in `ScreenTitleStepCount` to be a bit confusing. We have a data class for this, but it seems like we could just pass the string, imo. Also, and what makes it confusing, imo, is that the parameters for the constructor are passed in backwards from the perspective of the format string (i.e. `%1$d of %2$d`, followed by `stepsCount`, then `stepPosition`), only to then be switched back again when the data class instance is consumed. Though not at all part of this PR, I'm noting this as we may have an opportunity to clean some of this up - not suggesting any changes in this PR, though :smile:.

Reply:
> it seems to have been a part of this issue: https://github.com/wordpress-mobile/WordPress-Android/issues/11420 which interestingly enough, was a long ago removal of similar screens that we are now adding (though the designs and features we are planning for this are improved since then).

Thank you for your insights here, that's very useful and adds a bit more context surrounding the "history" of the new screen we are adding.
It feels to me that the whole set of changes we've planned for the onboarding flow are clearly an improvement, although for now it may seem a bit odd that we're adding back a previously removed step.

> Tbh, I find the current logic in ScreenTitleStepCount to be a bit confusing. We have a data class for this, but it seems like we could just pass the string, imo. Also, and what makes it confusing, imo, is that the parameters for the constructor are passed in backwards from the perspective of the format string (i.e. %1$d of %2$d, followed by stepsCount, then stepPosition), only to then be switched back again when the data class instance is consumed. Though not at all part of this PR, I'm noting this as we may have an opportunity to clean some of this up - not suggesting any changes in this PR, though 😄.

Really nice finding here, this can definitely be improved to make things more clear.
Let's keep it in mind for a separate PR 🙋🏻‍♂️